# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/nsIQuotaManagerService.idl
# Commit: aaa63956f3f0
# Full Hash: aaa63956f3f022575b29059d5909f63079723d3a
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-10 09:22:33
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add nsIQuotaManagerService::PersistentOriginInitialized and nsIQuotaManagerService::TemporaryOriginInitialized methods; r=dom-storage-reviewers,jari
#   
#   This additional testing infrastructure is needed for verifying origin
#   initialization status.
#   
# ==============================================================================

diff -r f799f35b1c84 -r aaa63956f3f0 dom/quota/nsIQuotaManagerService.idl
--- a/dom/quota/nsIQuotaManagerService.idl	Wed Oct 09 22:16:50 2024 +0000
+++ b/dom/quota/nsIQuotaManagerService.idl	Wed Oct 09 22:16:50 2024 +0000
@@ -43,6 +43,34 @@
   temporaryStorageInitialized();
 
   /**
+   * Check if persistent origin is initialized.
+   *
+   * If the dom.quotaManager.testing preference is not true the call will be
+   * a no-op.
+   *
+   * @param aPrincipal
+   *        A principal for the persistent origin whose state is to be checked.
+   */
+  [must_use] nsIQuotaRequest
+  persistentOriginInitialized(in nsIPrincipal aPrincipal);
+
+  /**
+   * Check if temporary origin is initialized.
+   *
+   * If the dom.quotaManager.testing preference is not true the call will be
+   * a no-op.
+   *
+   * @param aPersistenceType
+   *        A string that tells what persistence type will be checked (either
+   *        "temporary" or "default").
+   * @param aPrincipal
+   *        A principal for the temporary origin whose state is to be checked.
+   */
+  [must_use] nsIQuotaRequest
+  temporaryOriginInitialized(in ACString aPersistenceType,
+                             in nsIPrincipal aPrincipal);
+
+  /**
    * Initializes storage directory. This can be used in tests to verify
    * upgrade methods.
    *